File: src/com/sleazyweasel/applescriptifier/PlaylistComboBoxModel.java
package com.sleazyweasel.applescriptifier;

import de.felixbruns.jotify.media.Playlist;

import javax.swing.*;
import java.util.List;
import java.util.Vector;
import java.util.logging.Logger;

public class PlaylistComboBoxModel extends DefaultComboBoxModel {
    private static final Logger logger = Logger.getLogger(PlaylistComboBoxModel.class.getName());
    private final NativeSpotifySupport spotifySupport;

    public PlaylistComboBoxModel(NativeSpotifySupport spotifySupport) {
        super(new Vector<Playlist>());
        this.spotifySupport = spotifySupport;
        setSelectedItem(null);
    }

    public Playlist getSelectedStation() {
        return (Playlist) super.getSelectedItem();
    }

    public void refreshContents() {
        List<Playlist> playlists = spotifySupport.getPlaylists();
        Playlist selectedItem = getSelectedStation();
        logger.info("selectedItem = " + selectedItem);
        removeAllElements();
        for (Playlist playlist : playlists) {
            addElement(playlist);
        }
        setSelectedItem(selectedItem);
    }
}

File: src/com/sleazyweasel/applescriptifier/PlaylistComboBoxModel.scala
package com.sleazyweasel.applescriptifier

import de.felixbruns.jotify.media.Playlist
import javax.swing._
import java.util.logging.Logger
import java.util

class PlaylistComboBoxModel(val spotifySupport: NativeSpotifySupport) extends DefaultComboBoxModel(new util.Vector[Playlist]) {
  private final val logger: Logger = Logger.getLogger(classOf[PlaylistComboBoxModel].getName)

  setSelectedItem(null)

  def getSelectedStation: Playlist = {
    super.getSelectedItem.asInstanceOf[Playlist]
  }

  def refreshContents() {
    val playlists: util.List[Playlist] = spotifySupport.getPlaylists
    val selectedItem: Playlist = getSelectedStation
    logger.finest("selectedItem = " + selectedItem)
    removeAllElements()
    import scala.collection.JavaConversions._
    for (playlist <- playlists) {
      addElement(playlist)
    }
    setSelectedItem(selectedItem)
  }

}